DOC: Fixed documentation for few files

ShaharNaveh opened the pull request:
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Commit:
Fix docs for algorithms.py

Commit:
Fix docs for nanops.py

Commit:
Fix docs for indexers.py

ShaharNaveh commented on pandas/core/indexers.py:
```diff
@@ -209,16 +209,21 @@ def validate_indices(indices: np.ndarray, n: int) -> None:
+        ...
+    ValueError: negative dimensions are not allowed
+
+    >>> validate_indices(np.ndarray([0, 1]), 0) # OK
```
The docs in master ATM says it should raise an ```IndexError```, but it does not.
Would be glad if someone will take a look here

phofl replied:
Why are you using np.ndarray instead of np.array? The latter one raises

ShaharNaveh replied:
> Why are you using np.ndarray instead of np.array? The latter one raises

I complety missed it, I have pushed a fix for it in https://github.com/pandas-dev/pandas/pull/40903/commits/857c11060a30191c4001188324fc50080dc22c07

MarcoGorelli commented:
> dont' we have a master issue for this? (e.g. getting all doc-tests running)? cc @MarcoGorelli

I wasn't aware that not all doctests were running - anyway, there's #22459 still open

jreback commented on ci/code_checks.sh:
```diff
@@ -138,6 +138,10 @@ if [[ -z "$CHECK" || "$CHECK" == "doctests" ]]; then
 
     # Individual files
 
+    MSG='Doctests algorithms.py' ; echo $MSG
```
can these be combined? e.g. 1 for files, 1 for directories

ShaharNaveh replied:
Sure, will open a separate PR for that.

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-Fix-algo-index-nan

Commit:
Switching from `np.ndarray` to `np.array`

As pointed here:
https://github.com/pandas-dev/pandas/pull/40903#discussion_r611884623

Commit:
Rerunning the CI

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-Fix-algo-index-nan

Commit:
Fixed failing tests

Commit:
Applied "black formatting" on the example code

phofl commented on pandas/core/algorithms.py:
```diff
@@ -375,46 +375,60 @@ def unique(values):
+    ...         ]
+    ...     )
+    ... )
     DatetimeIndex(['2016-01-01 00:00:00-05:00'],
```
Format here looks a bit odd?

ShaharNaveh replied:
That's the "Black" formatting.

I personally think it's easier on the eyes, but that's not a deal breaker for me.

Can revert if you think it's best to leave it as it was before

phofl replied:
Sorry if this was misleading, I was referring to the lines you haven‘t touched, they look out of sync now

ShaharNaveh replied:
I didn't managed to get the output lines to follow the "black" formatting styles.

What I could do is to make the output to appear in a single line, instead of multiple by using ```\```. i.e

```python
def foo():
    """
    Examples
    --------------
    >> foo()
    Line1 \
Line 2 \
Line 3 \
    """
```

Will result in the docs as:
```
>>> foo()
Line 1 Line 2 Line 3
```

phofl replied:
Sounds good to me

phofl commented:
small comment, otherwise lgtm

jreback reviewed (changes requested):
looks good, ex @phofl comment, ping on green

Commit:
Revert "Applied "black formatting" on the example code"

This reverts commit bec5b3ae5752efe4c7bd2efd9040fe0304592e39.

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-Fix-algo-index-nan

Commit:
Revert "Revert "Applied "black formatting" on the example code""

This reverts commit 6adfead166d4889f535415ed62c80272ae3ac5ae.

Commit:
Make output on single line

Commit:
Revert "Make output on single line"

This reverts commit f4d79c64a597cf6fec13e2176a143229d373e70e.

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-Fix-algo-index-nan

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-Fix-algo-index-nan

Commit:
Add the changed files to the CI test

jreback commented:
lgtm @phofl over to you

phofl commented:
thanks @ShaharNaveh
